instrumentation/CONTRIBUTING.md

```diff
@@ -0,0 +1,407 @@
+    * Provide executable examples in an `examples` directory
+5. Submit a pull request
+
+## Generate the Gem
+
+This repository contains a script to generate a new instrumentation library.
+
+The snippit below demonstrates how to generate a an instrumentation for the `werewolf` gem, starting from the repository root.
+
+```console
+
+$bash opentelemetry-ruby-contrib> bin/instrumentation_generator werewolf
+
+```
+
+The output of the generator shows that it creates a new directory in the `instrumentation` directory using the name of the gem:
+
+``` console
+
+🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨
+
+ WARNING: Your gem will *NOT* be tested until you add it to the CI workflows in `.github/workflows/ci.yml`!!
+
+🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨
+
+      create  instrumentation/werewolf/.rubocop.yml
+      create  instrumentation/werewolf/.yardopts
+      create  instrumentation/werewolf/Appraisals
+      create  instrumentation/werewolf/CHANGELOG.md
+      create  instrumentation/werewolf/Gemfile
+      create  instrumentation/werewolf/LICENSE
+      create  instrumentation/werewolf/opentelemetry-instrumentation-werewolf.gemspec
+      create  instrumentation/werewolf/Rakefile
+      create  instrumentation/werewolf/README.md
+      create  instrumentation/werewolf/lib/opentelemetry-instrumentation-werewolf.rb
+      create  instrumentation/werewolf/lib/opentelemetry/instrumentation.rb
+      create  instrumentation/werewolf/lib/opentelemetry/instrumentation/werewolf.rb
+      create  instrumentation/werewolf/lib/opentelemetry/instrumentation/werewolf/instrumentation.rb
+      create  instrumentation/werewolf/lib/opentelemetry/instrumentation/werewolf/version.rb
+      create  instrumentation/werewolf/test/test_helper.rb
+      create  instrumentation/werewolf/test/opentelemetry/instrumentation/werewolf/instrumentation_test.rb
+      insert  .toys/.data/releases.yml
+      insert  instrumentation/all/Gemfile
+      insert  instrumentation/all/opentelemetry-instrumentation-all.gemspec
+      insert  instrumentation/all/lib/opentelemetry/instrumentation/all.rb
+
+```
+
+## Implementation guidelines
+
+The original design and implementation of this project was heavily influenced by Datadog's `dd-trace-rb` project. You may refer to the [Datadog Porting Guide](datadog-porting-guide.md) as a reference for implementing instrumentations, however the following guidelines are specific to OpenTelemetry Ruby:
+
+*. Use `OpenTelemetry::Instrumentation::Base`
+*. Use the OpenTelemetry API
+*. Use First Party Extension Points
+*. Use Semantic Conventions
+*. Write comprehensive automated tests
+*. Understand Performance Characteristics
+
+### Use `OpenTelemetry::Instrumentation::Base`
+
+The entry point of your instrumentation should be implemented as a subclass of `OpenTelemetry::Instrumentation::Base`:
+
+* Implement `install` block, where all of the integration work happens
+* Implement `present` block
```

Comment:
I think we need a very short description of what the `present` block expects, just like we did for `install` and `compatible`.

Reply:
Added some more details with a code example